### PR TITLE
Fix make sphinx failure on Windows

### LINKED ISSUE #9818: "make sphinx" fails on Windows upon starting coqtop
#### Description of the problem

`make sphinx` fails on Windows with the message from Python/expect `Cannot find a physical path bound to logical path matching suffix\r\r\n<> and prefix Coq.\r\r\n\r\r\n` upon the first invocation of coqtop.

Removing the COQLIB on this line in Makefile.doc fixes the problem on Windows.

```
SPHINXENV:=COQBIN="$(CURDIR)/bin/" COQLIB="$(CURDIR)"
```

Here's what you get running coqtop with COQLIB set by hand:
```
$ pwd
/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq

$ COQLIB="/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq"

$ export COQLIB

$ bin/coqtop.exe
Welcome to Coq 8.10+alpha (March 2019)
Warning:
Cannot open /cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq/theories
[cannot-open-path,filesystem]
Warning:
Cannot open /cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq/plugins
[cannot-open-path,filesystem]
Error: Cannot find a physical path bound to logical path matching suffix '<>' and prefix Coq.
```

#### Coq Version 8.9+

### PR DESCRIPTION
**Pull request by jfehrle**
**Kind:** infrastructure.

Fixes / closes #9818

### CODE REVIEW

**Comment by Zimmi48** [comment]
It seems like the change did not break the normal Linux build but it did break the build on Nix. I have no idea how this Makefile works, so I can't really help here.

**Comment by jfehrle** [comment]
@ejgallego Can you review/merge this small fix sometime soon?
Or perhaps @Zimmi48?

**Comment by ejgallego** [comment]
I am not the right person for this; maybe @gares ?

**Comment by gares** [comment]
Why not calling cygpath -w as you do in the bugreport? I don't recall why we set Coqlib, but there must have been a reason, so I'd rather keep it but fix the format.

**Comment by Zimmi48** [comment]
Indeed, not me. I'd gladly delegate to @gares here.

File: Makefile.doc [hunk 1]
```diff
@@ -31,7 +31,13 @@ DVIPS:=dvips
 HTMLSTYLE:=coqremote
 
 # Sphinx-related variables
+OSNAME:=$(shell uname -o)
+ifeq ($(OSNAME),Cygwin)
+# workaround awaiting fix to https://github.com/coq/coq/issues/9799 including coqtop
+SPHINXENV:=COQBIN="$(CURDIR)/bin/"
```
Comment (gares) [inline Makefile.doc]: I think we should keep the COQLIB, but fix its format by callyng `cygpath -w`

**Review by gares — CHANGES_REQUESTED** [review]

**Comment by SkySkimmer** [comment]
This seems like the same issue that led to using ocaml_pwd in the test suite, should we use it here too?

**Comment by gares** [comment]
You mean, for the value of `CURDIR` ?

**Comment by SkySkimmer** [comment]
yes

Gaëtan Gilbert

**Comment by jfehrle** [comment]
> This seems like the same issue that led to using ocaml_pwd in the test suite, should we use it here too?

Good thought, but no, ocaml_pwd generates a Cygwin path when a Windows path is needed.

```
$ ocaml test-suite/ocaml_pwd.ml .
/cygdrive/c/Users/Jim.DESKTOP-MMPNVGH/Documents/coqSource/coq
```

@gares Please take another look after the build completes.

**Comment by gares** [comment]
LGTM

**Review by gares — APPROVED** [review]

**Pull request merged**